pkg/config/v2_1_16/tidb.go

```diff
@@ -0,0 +1,319 @@
+	MaxTxnTimeUse uint `toml:"max-txn-time-use" json:"max-txn-time-use"`
+}
+
+// Plugin is the config for plugin
+type Plugin struct {
+	Dir  string `toml:"dir" json:"dir"`
+	Load string `toml:"load" json:"load"`
+}
+
+// Binlog is the config for binlog.
+type Binlog struct {
+	Enable       bool   `toml:"enable" json:"enable"`
+	WriteTimeout string `toml:"write-timeout" json:"write-timeout"`
+	// If IgnoreError is true, when writting binlog meets error, TiDB would
+	// ignore the error.
+	IgnoreError bool `toml:"ignore-error" json:"ignore-error"`
+	// Use socket file to write binlog, for compatible with kafka version tidb-binlog.
+	BinlogSocket string `toml:"binlog-socket" json:"binlog-socket"`
+}
+
+var defaultTidbConf = TidbConfig{
+	Host:                         "0.0.0.0",
+	AdvertiseAddress:             "",
+	Port:                         4000,
+	Store:                        "mocktikv",
+	Path:                         "/tmp/tidb",
+	RunDDL:                       true,
+	SplitTable:                   true,
+	Lease:                        "45s",
+	TokenLimit:                   1000,
+	OOMAction:                    "log",
+	MemQuotaQuery:                32 << 30,
+	EnableStreaming:              false,
+	EnableBatchDML:               false,
+	CheckMb4ValueInUTF8:          true,
+	TreatOldVersionUTF8AsUTF8MB4: true,
+	TxnLocalLatches: TxnLocalLatches{
+		Enabled:  false,
+		Capacity: 10240000,
+	},
+	LowerCaseTableNames: 2,
+	Log: Log{
+		Level:              "info",
+		Format:             "text",
+		File:               NewFileLogConfig(true, DefaultLogMaxSize),
+		SlowQueryFile:      "tidb-slow.log",
+		SlowThreshold:      DefaultSlowThreshold,
+		ExpensiveThreshold: 10000,
+		QueryLogMaxLen:     DefaultQueryLogMaxLen,
+	},
+	Status: Status{
+		ReportStatus:    true,
+		StatusPort:      10080,
+		MetricsInterval: 15,
+	},
+	Performance: Performance{
+		TCPKeepAlive:        true,
+		CrossJoin:           true,
+		StatsLease:          "3s",
+		EnableUpdateStats:   true,
+		RunAutoAnalyze:      true,
+		StmtCountLimit:      5000,
+		FeedbackProbability: 0.05,
+		QueryFeedbackLimit:  1024,
+		PseudoEstimateRatio: 0.8,
+		ForcePriority:       "NO_PRIORITY",
+		TxnEntryCountLimit:  DefTxnEntryCountLimit,
+		TxnTotalSizeLimit:   DefTxnTotalSizeLimit,
+	},
+	XProtocol: XProtocol{
+		XHost: "",
+		XPort: 0,
+	},
+	ProxyProtocol: ProxyProtocol{
+		Networks:      "",
+		HeaderTimeout: 5,
+	},
+	PreparedPlanCache: PreparedPlanCache{
+		Enabled:  false,
+		Capacity: 100,
+	},
+	OpenTracing: OpenTracing{
+		Enable: false,
+		Sampler: OpenTracingSampler{
+			Type:  "const",
+			Param: 1.0,
+		},
+		Reporter: OpenTracingReporter{},
+	},
+	TiKVClient: TiKVClient{
+		GrpcConnectionCount:  16,
+		GrpcKeepAliveTime:    10,
+		GrpcKeepAliveTimeout: 3,
+		CommitTimeout:        "41s",
+		MaxTxnTimeUse:        590,
+	},
+	Binlog: Binlog{
+		WriteTimeout: "15s",
+	},
+}
+
+var globalConf = defaultTidbConf
```

Comment:
This var is not used by any package

Reply:
The configuration file provided by users usually are part of the configuration, other values are using default values defined by PD/TiKV/TiDB itself. Some of default values may still be important. So this variable is kept for future use for more detailed check.